Implemented token based authentication for django rest framework

shreygoel7 opened the pull request.

shreygoel7 committed:
Implemented token based authentication for django rest framework

dynamoh reviewed (changes requested):
REST API views and HttpResponse views should be kept separate.
Folder structure should be :
```
- globals
    - api
         - views.py
         - urls.py
         - serializers.py
```

garg3133 commented:
@shreygoel7 Can you please remove all the migrations from this PR (and also delete them from your system). I've fixed all the migration issues on Fusion and merged them in #280.
(We need to take extra care regarding the migrations otherwise it may cause problems in the database. We'll soon release the guidelines for making migrations).

Steps to be followed:
* Delete all the 7 migration files you created and pushed in this PR.
* Stage them (`git add <name of 7 files>` or use Git tab in vscode to do this using GUI).
* Run `git commit --amend`. (It will edit the previous commit you made so make sure you didn't commit anything after making this PR).
* Now force push your modified commit to this PR (`git push origin token-based-auth`).

Also, please don't run migrations command on your system. Whenever you make any changes in the models and want to migrate them, commit the changed models files and create a PR and I will migrate those changes for you (it will ensure consistency among migrations and will keep the database safe).

Let me know if you face any problem :)

garg3133 commented on FusionIIIT/Fusion/settings/development.py:
```diff
@@ -16,6 +16,15 @@
     }
 }
 
+REST_FRAMEWORK = {
```
Doesn't this needs to be added in productions settings also?

shreygoel7 replied:
Added in productions settings.

garg3133 reviewed (commented):
Also as @dynamoh said, you should write the views and urls for API in separate files.

views should be written in `globals/api/views.py` and urls in `globals/api/urls.py` files.

shreygoel7 committed:
Implemented DRF token based login api.

shreygoel7 committed:
Merge branch 'token-based-auth' of https://github.com/shreygoel7/Fusion into token-based-auth

shreygoel7 committed:
Removing migrations

shreygoel7 commented:
Made all the changes

dynamoh reviewed (changes requested):
- [x] change base route from ```/api/``` to ```/api/auth/``` for auth APIs.
- [x] Instead of GenericViewset create an APIView for Login.

shreygoel7 committed:
Changed:
1. GenericViewset to APIView for login api.
2. Base route from /api/ to /api/auth/
